### PR TITLE
feat: send sign info

### PR DESCRIPTION
**Pull request by 0xAaCE**
# feat: send sign info

## Summary

We gonna ask sign validation in some cases so we need get the info from provider

### CODE REVIEW

File: src/Provider.ts [hunk 1]
```diff
@@ -79,30 +81,84 @@ export interface ProviderInterface {
...
-    const res = await clientRPC.call("sign", [payloadArr, metadata], {
-      timeout: 0,
-      target: "",
-    });
+
+    if (signInfo) signInfo.decodedArguments = decodeArgs(signInfo, argsTypes);
+
+    const res = await clientRPC.call(
+      "requestSign",
+      [payloadArr, metadata, signInfo],
```
Comment (tomiir) [inline src/Provider.ts]: All of this should be in a separate file

File: src/Provider.ts [hunk 2]
```diff
@@ -79,30 +81,84 @@ export interface ProviderInterface {
...
-type CallConfigObject = {
-  timeout?: number;
-  target?: string;
+export interface AssuredSignInfo {
+  methodName: string;
+  requestType: string;
+  canisterId: string;
+  sender: string;
+  arguments: Buffer;
+  decodedArguments?: JsonValue;
```
Comment (tomiir) [inline src/Provider.ts]: what does JsonValue imply? it's like an `any` json?

File: src/Provider.ts [hunk 3]
```diff
@@ -79,30 +81,84 @@ export interface ProviderInterface {
...
+  return !!(
+    signInfo?.canisterId &&
+    signInfo?.methodName &&
+    signInfo?.arguments &&
+    argsTypes[signInfo.canisterId]?.[signInfo.methodName]
+  );
+};
+
+const decodeArgs = (signInfo: SignInfo, argsTypes: ArgsTypesOfCanister) => {
+  if (canDecodeArgs(signInfo, argsTypes)) {
```
Comment (tomiir) [inline src/Provider.ts]: what if you can't decode them? that's the direct agent case right?

Comment (0xAaCE) [reply src/Provider.ts]: Yep it is, if you cant, requestInfo which the FE receive not gonna have decodedArguments nothing else

File: src/Provider.ts (hunk 1 above)
Comment (0xAaCE) [reply src/Provider.ts]: Good idea 👍

File: src/Provider.ts (hunk 2 above)
Comment (0xAaCE) [reply src/Provider.ts]: It is an type fron @dfinity libraries and i think it is something like any type that could be json.stringify. But i am not sure

**Review by maksym-boytsov — APPROVED** [review]

**Pull request merged**